File: KwangHo/영어 끝말잇기.py
def solution(n, words):
    answer = []

    arr=[]
    a=0
    b=0
    
    arr.append(words[0])
    for i in range(1,len(words)):
        t = words[i]
        if len(t) > 1 and words[i-1][-1] == t[0] and t not in arr :
            arr.append(t)
        else :
            a = i%n +1
            b = i//n +1
            break
    
    answer.append(a)
    answer.append(b)
    return answer

'''
[조건]
1. 길이 1보다 큰지
2. 끝단어 == 첫단어
3. 중복체크
이외라면 틀렷으므로 출력
'''

File: Doodream/영어 끝말잇기.js
function solution(n, words) {
    let answer = [];
    let number = 0;
    let order = 0;

    for (let i = 1; i < words.length; i++) {
        let subArr = words.slice(0, i);
        if ((words[i - 1].substring(words[i - 1].length - 1) !== words[i][0]) || (subArr.some(word => word === words[i]))) {
            if ((i + 1) % n === 0) {
                number = n;
            } else {
                number = (i + 1) % n;
            }
            order = Math.ceil((i + 1) / n);
            break;
        }
    }
    if (number === 0 && order === 0) {
        answer = [0, 0];
    } else {
        answer = [number, order];
    }

    return answer;
}